src/web.ts

```diff
@@ -26,13 +26,17 @@ export class SecureStoragePluginWeb extends WebPlugin implements SecureStoragePl
     return Promise.resolve({ value: true });
   }
   clear(): Promise<{ value: boolean }> {
-    for (var key in localStorage) {
+    for (const key in localStorage) {
       if (key.indexOf(this.PREFIX) === 0) {
         localStorage.removeItem(key);
       }
     }
     return Promise.resolve({ value: true });
   }
+  keys(): Promise<{ value: string[] }> {
+    const keys = Object.keys(localStorage).filter(k => k.indexOf(this.PREFIX) === 0);
```

Comment:
Thank you for PR! Is someting like this used in Android and iOS? From just a quick review it seems keys() method would return all entries from sharedpreferences file used for whole application, so it may contain keys which are not created by this plugin. The same seems to apply for iOS and keychain (I am not sure).

Reply:
Thank you for your quick response. And good point - I'll take a look if we can and should prefix the keys in ios and android like it's done currently in the web implementation and update this PR.

Reply:
ios is now using a dedicated keychain wrapper service by [defining a name](https://github.com/matthiasgrube/capacitor-secure-storage-plugin/commit/18856dcf0008517e616494eb68c22b006f0ead2a#diff-6090fa5cc942a713e2be00617c512a9d459f905c3ea1091ca79031b1a69146b7R11).
From the docs:
_The ServiceName for this instance. Used to uniquely identify all keys stored using this keychain wrapper instance._

android is already using a custom preferences file:
[SKD16](https://github.com/martinkasa/capacitor-secure-storage-plugin/blob/master/android/src/main/java/com/whitestein/securestorage/PasswordStorageHelper.java#L103)
[SKD18](https://github.com/martinkasa/capacitor-secure-storage-plugin/blob/master/android/src/main/java/com/whitestein/securestorage/PasswordStorageHelper.java#L153)

Reply:
Thanks for quick fixes. Using dedicated keychain wrapper is a breaking change, so users who already use this plugin would not be able to access their stored data. Do you have any suggestion regarding this?

Reply:
I can only think of a one time migration to prevent data loss when switching to the dedicated keychain wrapper:

1. test if migration flag is present in dedicated keychain wrapper on plugin initialization (if yes, go to step 4)
2. copy values from standard ios keychain wrapper to dedicated keychain wrapper (via the new keys() and the existing get() methods)
3. add migration flag to dedicated keychain wrapper after copying the data
4. use dedicated keychain wrapper

I would not cleanup the standard keychain (as you wrote, this could have bad side effects). Sadly this would also copy the whole standard keychain to the dedicated one.

Can you think of any other way to do the migration without data loss?

Reply:
It sounds good. keys() in old mode would return all keys, and can migrate just the ones that have 'cap_sec' prefix.

But actually I like second idea with read from standard keychain as fallback more. Simple and no migration needed.

Reply:
on ios the standard keychain wrapper is now used as a fallback for the getKey method. I've also extended the test cases to ensure the dedicated wrapper does not modify anything in the standard one.